src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1839,6 +1882,8 @@ private string GetRedPropertyType(Field field)
         {
             if (field.Type == "SyntaxList<SyntaxToken>")
                 return "SyntaxTokenList";
+            if (IsOptional(field) && IsNode(field.Type) && field.Type != "SyntaxToken")
```

Comment:
Maybe IsOptionalReferenceType helper?

Reply:
I'm not sure it matters in this internal tool. I'll leave this type of feedback for the core compiler team.